tests/test_parser.py

```diff
@@ -50,19 +50,84 @@ def test_no_scenario(self, test_parser: Mock, parser: Parser):
-            fake_tests.append(Mock())
-            fake_tests[-1].name = f"test-{i}"
-        test_parser.return_value = fake_tests
+
+        fake_tests = [Mock(name=f"test-{i}") for i in range(3)]
+        for i, test in enumerate(fake_tests):
+            test.name = f"test-{i}"
+
+        test_parser.side_effect = [fake_tests, []]
+
         fake_scenario = Mock()
         fake_scenario.test_runs = [Mock()]
         fake_scenario.test_runs[0].test.name = "test-1"
         test_scenario_parser.return_value = fake_scenario
+
         _, tests, _ = parser.parse(tests_dir, Path())
+
         assert len(tests) == 1
+        assert tests[0].name == "test-1"
+
+    @patch("cloudai._core.test_parser.TestParser.parse_all")
+    @patch("cloudai._core.test_scenario_parser.TestScenarioParser.parse")
+    @patch("cloudai.parser.Parser.parse_plugins")
+    def test_scenario_with_plugin_common_tests(
+        self, parse_plugins: Mock, test_scenario_parser: Mock, test_parser: Mock, parser: Parser
+    ):
+        tests_dir = parser.system_config_path.parent.parent / "test"
+
+        main_tests = [Mock() for _ in range(3)]
+        for i, test in enumerate(main_tests):
+            test.name = f"test-{i}"
+        plugin_tests = [Mock()]
+        plugin_tests[0].name = "test-1"
+
+        test_parser.side_effect = [main_tests, plugin_tests]
+
+        fake_scenario = Mock()
+        fake_scenario.test_runs = [Mock()]
+        fake_scenario.test_runs[0].test.name = "test-1"
+        test_scenario_parser.return_value = fake_scenario
+
+        fake_plugin = Mock()
+        fake_plugin.test_runs = [Mock()]
+        fake_plugin.test_runs[0].test.name = "test-1"
+        parse_plugins.return_value = {"plugin-1": fake_plugin}
+
+        _, tests, _ = parser.parse(tests_dir, Path())
+
+        filtered_test_names = {t.name for t in tests}
+        assert len(tests) == 1
+        assert "test-1" in filtered_test_names
+
+    @patch("cloudai._core.test_parser.TestParser.parse_all")
+    @patch("cloudai._core.test_scenario_parser.TestScenarioParser.parse")
+    def test_scenario_with_plugin_exclusive_tests(self, test_scenario_parser: Mock, test_parser: Mock, parser: Parser):
+        tests_dir = parser.system_config_path.parent.parent / "test"
+        test_scenario_path = Path("/mock/test_scenario.toml")
+
+        main_tests = [Mock() for _ in range(3)]
+        plugin_tests = [Mock()]
+        for i, test in enumerate(main_tests):
+            test.name = f"test-{i}"
+        plugin_tests[0].name = "plugin-test-1"
+
+        test_parser.side_effect = [main_tests, plugin_tests]
+
+        fake_scenario = Mock()
+        fake_scenario.test_runs = [Mock()]
+        fake_scenario.test_runs[0].test.name = "test-1"
+        test_scenario_parser.return_value = fake_scenario
+
+        _, filtered_tests, _ = parser.parse(tests_dir, test_scenario_path)
+
+        filtered_test_names = {t.name for t in filtered_tests}
+        assert len(filtered_tests) == 2
```

Comment:
[minor] IMO it is OK to simplify to have only one regular test and one plugin.

Reply:
It already has only one test and one plugin. As a result, we need two filtered tests (one for regular use, one for the plugin). Initially, I added three tests to verify that only the regular tests are filtered.

Reply:
Ah, sorry, missed that two assert go with "not in"